### PR TITLE
[Mac] Extension support work.  TodayExtension runs, FinderSync templa…

### PR DESCRIPTION
**Pull request by timrisi**
…te added but not runnable yet.

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build success

File: msbuild/Xamarin.Mac.Tasks/Tasks/Mmp.cs [hunk 1]
```diff
@@ -163,8 +165,9 @@ protected override string GenerateCommandLineCommands ()
 					args.AddQuoted ("/assembly:" + Path.GetFullPath (asm));
 			}
 
-			if (!string.IsNullOrEmpty (ApplicationAssembly))
-				args.AddQuoted (Path.GetFullPath (ApplicationAssembly));
+			if (!string.IsNullOrEmpty (ApplicationAssembly)) {
+				args.AddQuoted(Path.GetFullPath (ApplicationAssembly + (IsAppExtension ? ".dll" : ".exe")));
```
Comment (rolfbjarne) [inline msbuild/Xamarin.Mac.Tasks/Tasks/Mmp.cs]: Style: space before opening parenthesis.

File: runtime/launcher.m [hunk 2]
```diff
@@ -429,29 +431,45 @@
...
+        
+        MonoAssembly *assembly = xamarin_open_assembly ("Xamarin.Mac.dll");
+        if (!assembly)
+            xamarin_assertion_message ("Failed to load %s.", "Xamarin.Mac.dll");
+        
+        MonoImage *image = mono_assembly_get_image (assembly);
+        
+        MonoClass *app_class = mono_class_from_name (image, "AppKit", "NSApplication");
+        if (!app_class)
+            xamarin_assertion_message ("Fatal error: failed to load the NSApplication.Init");
```
Comment (rolfbjarne) [inline runtime/launcher.m]: The assertion message is wrong, it says a method failed to load, but at this point it's the class that failed to load.

Maybe this assertion should go just below, when you get the method?

File: runtime/launcher.m [hunk 3]
```diff
@@ -558,11 +576,26 @@ int main (int argc, char **argv)
...
+        
 		free (new_argv);
 	}
 
 	return rv;
 }
+
+int main (int argc, char **argv)
+{
+    return xamarin_main (argc, argv, false);
```
Comment (rolfbjarne) [inline runtime/launcher.m]: Space indentation (should be tab).

File: runtime/launcher.m [hunk 4]
```diff
@@ -558,11 +576,26 @@ int main (int argc, char **argv)
...
 			*ptr++ = argv [i];
 		*ptr = NULL;
-
-		rv = mono_main (new_argc, new_argv);
-
+        
+		if (is_extension) {
+	            rv = NSExtensionMain (new_argc, new_argv);
+	        }
+	        else {
```
Comment (rolfbjarne) [inline runtime/launcher.m]: The `else` clause should go with the closing brace on the line above.

File: runtime/launcher.m (hunk 3 above)
Comment (rolfbjarne) [inline runtime/launcher.m]: Actually it looks like all your indentation in this file is spaces, and not tabs (wrong editor configuration maybe?)

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request closed without merging**